luaui/Widgets/cmd_mex_snap.lua

```diff
@@ -369,38 +381,60 @@ function widget:DrawWorld()
 	gl.DepthTest(true)
 
 	-- Add/update unit shape rendering
-	local newUnitshape = {-activeCmdID, bestPos.x, bestPos.y, bestPos.z, bface}
-	if not unitshape or (unitshape[1]~= newUnitshape[1] or unitshape[2]~= newUnitshape[2] or unitshape[3]~= newUnitshape[3] or unitshape[4]~= newUnitshape[4] or unitshape[5]~= newUnitshape[5]) then
+	local newUnitshape = { -activeCmdID, bestPos.x, bestPos.y, bestPos.z, bface }
+	if
+		not unitshape
+		or (
+			unitshape[1] ~= newUnitshape[1]
+			or unitshape[2] ~= newUnitshape[2]
+			or unitshape[3] ~= newUnitshape[3]
+			or unitshape[4] ~= newUnitshape[4]
+			or unitshape[5] ~= newUnitshape[5]
+		)
+	then
 		clearShape()
 		unitshape = newUnitshape
-		unitshape[6] = WG.DrawUnitShapeGL4(unitshape[1], unitshape[2], unitshape[3], unitshape[4], unitshape[5]*math_pi, 0.66, spGetMyTeamID(), 0.15, 0.3)
+		unitshape[6] = WG.DrawUnitShapeGL4(
+			unitshape[1],
+			unitshape[2],
+			unitshape[3],
+			unitshape[4],
+			unitshape[5] * math_pi,
+			0.66,
+			spGetMyTeamID(),
+			0.15,
+			0.3
+		)
 	end
 end
 
 local lastMexCmd = 0
 function widget:CommandNotify(cmdID, cmdParams, cmdOpts)
-
 	local metalExtracts = isMex[-cmdID]
 
-	if not metalExtracts then return end
+	if not metalExtracts then
+		return
+	end
 
-	if cmdOpts.mexsnap then return end -- notifying order
+	if cmdOpts.mexsnap then
+		return
+	end -- notifying order
 
-	if lastMexCmd+0.1 > os.clock() then return end	-- ignore accidental line-drag cmd
+	if lastMexCmd + 0.1 > os.clock() then
+		return
+	end -- ignore accidental line-drag cmd
 	lastMexCmd = os.clock()
 
-	local cbx, cbz = cmdParams[1], cmdParams[3]
-
 	local orders = cmdOpts.shift and GetClashingOrders() or {}
-	local closestSpot = GetClosestMex(bx, bz, WG['resource_spot_finder'].metalSpotsList, metalExtracts, orders)
+	local closestSpot = GetClosestMex(bx, bz, metalSpotsList, metalExtracts, orders)
 
-	if closestSpot then --and not WG['resource_spot_finder'].IsMexPositionValid(closestSpot, cbx, cbz) then
+	if closestSpot then --and not IsMexPositionValid(closestSpot, cmdParams[1], cmdParams[3]) then
```

Comment:
I kept it commented, but original had this check which was commented in the regression commit, didn't care enough to dig the edge case I used it to address.